wallets/vault.js

```diff
@@ -0,0 +1,134 @@
+import { getWalletByType } from '@/wallets/common'
+import walletDefs from '@/wallets/client'
+import { get } from '@/lib/object'
+
+// returns fragments for the Prisma Client API
+// input is a wallet row from the db with the new schema
+export function vaultPrismaFragments (wallet) {
```

Comment:
It might make sense to refactor this into individual, independent parts.

Reply:
I would have loved to write a unit tests for this so that I don't accidentally change the output between now and after my refactor but I can't because of NDK's dependency on Cashu ...

![2025-04-20-235434_1914x1163_scrot](https://github.com/user-attachments/assets/6419366f-612d-4e9e-8e2f-b88a1cbb43b2)